src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/MacProxy.cs

```diff
@@ -0,0 +1,125 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Http;
+using System.Net;
+using System.Collections.Generic;
+using Microsoft.Win32.SafeHandles;
+using static Interop.CoreFoundation;
+using static Interop.RunLoop;
+
+using CFRunLoopRef = System.IntPtr;
+
+namespace System.Net.Http
+{
+    internal sealed class MacProxy : IWebProxy
+    {
+        public ICredentials Credentials
+        {
+            get => null;
+            set => throw new NotSupportedException();
+        }
+
+        static Uri GetProxyUri(string scheme, CFProxy proxy)
+        {
+            var uriBuilder = new UriBuilder(
+                scheme,
+                proxy.HostName,
+                proxy.PortNumber);
+
+            // FIXME: Handle user name and password
+
+            return uriBuilder.Uri;
+        }
+
+        public Uri ExecuteProxyAutoConfiguration(SafeCreateHandle cfurl, CFProxy proxy)
```

Comment:
What kind of script is this?  JavaScript.  Running JavaScript in-process is something we did a long time ago in .NET Framework but stopped that due to various problems.  On Windows, this is handled by an out-of-process service.

I don't think we want to be executing scripts in process that are not validated.

Reply:
It is JavaScript, but it's executed using system APIs the same way as Chrome / Mono do it. It is not documented whether the APIs execute it in process or out of process, but they are designed to be somewhat asynchronous, so anything is possible.